content/docs/guides/revocation.md

```diff
@@ -65,6 +74,149 @@ check if time($date), $date <= 2018-12-20T00:00:00+00:00;
 
 ### Providing a revocation list during biscuit authorization
 
-#### In haskell
+* [in haskell](../../Usage/haskell/#reject-revoked-tokens)
+* [in rust](../../Usage/rust/#reject-revoked-tokens)
+
+## Why should we plan for token revocation?
+
+There are two competing approaches to session management in authorization, that will drive
+architectural decisions:
+- in *stateful* systems, all authorizations are performed through one service or database
+that holds the list of currently active sessions
+- in *stateless* systems, authorization can be performed independently in any service, only
+using information from the token and the service. In particular, the service cannot know
+about all of the currently active sessions (there may not even be a concept of session)
+
+Those two solutions are often compared on their ability to close a session. Why? Can't we
+just set an expiration date? Unfortunately, even an with expiration date we would still need a way to close
+a session, to implement the log out functionality. That feature is common, expected by users,
+and needed in multiple situations (ex: public computer, disconnecting sessions from a stolen
+phone). Even for purely service to service communication, we will need to close the access
+once the client service is decommissioned.
+
+In stateful systems, closing a session is easy: we delete the session's information from the
+database and that's it. In stateless systems, this is more complex: how do we make sure
+all services know that the session is invalid? That means reintroducing some shared state,
+so is the stateless design impossible after all? Shouldn't we go back to stateful systems?
+
+If the architecture we are designing can rely on a central state, it will be the simplest
+approach and probably the right solution. But there are good reasons to choose the stateless
+design:
+- **scaling**: a central authorization service that is queried on every request is a single point
+of failure for the entire system. If it is down, nobody can log in, but existing sessions will
+fail too. A stateless approach will decouple session creation from authorization, so existing
+sessions can still work when the authentication service fails.
+- **isolation**: the service receiving the request might be less trusted and should not be
+able to access session information.
+- **authentication delegation**: authentication could be in a separate service (example: SSO)
+that can't be queried on every request. That service could even be managed by a different
+company.
+
+In those cases, separating authorization from session creation makes sense, but then how do
+we close a session? It is usually done through token revocation: the authorizer needs to
+know a list of tokens that must be refused, and that list changes dynamically, so we are
+reintroducing some state in the system.
```

Comment:
This sentence is a repetition. From my Monkey POV, It is implicit hear

Reply:
the tricky parts need a lot of repetition :)

Reply:
👍